### PR TITLE
Add V4 repl console

### PR DESCRIPTION
**Pull request by mullermp**

### CODE REVIEW

File: gems/aws-sdk-core/lib/aws-sdk-core.rb [hunk 1]
```diff
@@ -9,60 +9,70 @@ module AWS
...
-            begin
-              require 'aws-sdk-sts'
-              true
-            rescue LoadError, NameError
-              false
-            end
+      class << self
+        # @return [Hash] Returns a hash of default configuration options shared
+        #   by all constructed clients.
+        def config
```
Comment (alextwoods) [inline gems/aws-sdk-core/lib/aws-sdk-core.rb:15]: Can we do this as a separate PR, it feels un-related to the repl and is a pretty big change.

File: aws-v4.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+gems = Dir['gems/*']
+gems.each do |gem_path|
+  $LOAD_PATH.unshift(File.expand_path('lib', gem_path))
+end
+# TODO: This is slow - we should autoload but it requires a complete
+# code generated list of services like aws-sdk-resources.
```
Comment (alextwoods) [inline aws-v4.rb:9]: I think we should potentially write out a services.json sort of file during codegeneration that we could use here.  The autoload is highly preferable.

File: gems/aws-sdk-core/lib/aws-sdk-core.rb (hunk 1 above)
Comment (mullermp) [reply gems/aws-sdk-core/lib/aws-sdk-core.rb]: yeah

File: aws-v4.rb (hunk 2 above)
Comment (mullermp) [reply aws-v4.rb:9]: yeah. out of scope right now ..
